packages/nimble-components/src/table/components/group-row/tests/table-group-row.spec.ts

```diff
@@ -0,0 +1,59 @@
+import { html } from '@microsoft/fast-element';
+import { TableGroupRow } from '..';
+import {
+    TableColumnText,
+    TableColumnTextColumnConfig
+} from '../../../../table-column/text';
+import { waitForUpdatesAsync } from '../../../../testing/async-helpers';
+import { fixture, Fixture } from '../../../../utilities/tests/fixture';
+import { TableGroupRowPageObject } from './table-group-row.pageobject';
+
+// prettier-ignore
+async function setup(): Promise<Fixture<TableGroupRow>> {
+    return fixture<TableGroupRow>(
+        html`<nimble-table-group-row>
+            </nimble-table-group-row>`
+    );
+}
+
+describe('TableGroupRow', () => {
+    let element: TableGroupRow;
+    let connect: () => Promise<void>;
+    let disconnect: () => Promise<void>;
+    let pageObject: TableGroupRowPageObject;
+
+    beforeEach(async () => {
+        ({ element, connect, disconnect } = await setup());
+        pageObject = new TableGroupRowPageObject(element);
+    });
+
+    afterEach(async () => {
+        await disconnect();
+    });
+
+    it('can construct an element instance', () => {
+        // prettier-ignore
+        expect(document.createElement('nimble-table-group-row')).toBeInstanceOf(TableGroupRow);
+    });
+
+    it('column state is applied to group header', async () => {
+        await connect();
+
+        const textColumn1 = new TableColumnText();
+        textColumn1.columnConfig = {
+            placeholder: 'no value'
+        } as TableColumnTextColumnConfig;
+        element.groupColumn = textColumn1;
+        element.groupRowValue = 'test';
+        await waitForUpdatesAsync();
+
+        const renderedGroupHeader = pageObject.getRenderedGroupHeader();
+
+        expect(renderedGroupHeader).not.toBeNull();
+        expect(renderedGroupHeader?.groupHeaderValue).toBe('test');
+        expect(
```

Comment:
nit: I think we should try to follow the best practices we established in `systemlink-lib-angular` page objects when we build them in Nimble. This would mean not exposing a public method to get the entire header, but rather making public methods for each of these attributes which return primitive types. It won't make much difference here yet, but if we start making page objects part of our public API then it will reduce the risk of breaking clients if they return primitive types.

Reply:
So, I've decided to remove the page object for now. The utility we were getting from it was minimal at the moment, as the facets it would have been testing are already being tested (via integration tests admittedly) in the `table-grouping.spec` set of tests. I have added a new test, however, to ensure we are firing the expected toggle event upon click.